quickstart.py

```diff
@@ -0,0 +1,70 @@
+from numpy import pi, log
+import pypolychord
+from pypolychord.priors import UniformPrior
+try:
+    from mpi4py import MPI
+except ImportError:
+    pass
+
+
+#| Define a four-dimensional spherical gaussian likelihood,
+#| width sigma=0.1, centered on the 0 with one derived parameter.
+#| The derived parameter is the squared radius
+
+nDims = 4
+nDerived = 1
+sigma = 0.1
+
+def likelihood(theta):
+    """ Simple Gaussian Likelihood"""
+
+    nDims = len(theta)
+    r2 = sum(theta**2)
+    logL = -log(2*pi*sigma*sigma)*nDims/2.0
+    logL += -r2/2/sigma/sigma
+
+    return logL, [r2]
+
+#| Define a box uniform prior from -1 to 1
+
+def prior(hypercube):
+    """ Uniform prior from [-1,1]^D. """
+    return UniformPrior(-1, 1)(hypercube)
+
+#| Optional dumper function giving run-time read access to
+#| the live points, dead points, weights and evidences
+
+def dumper(live, dead, logweights, logZ, logZerr):
+    print("Last dead point:", dead[-1])
+
+#| Parameter names
+#! This is a list of tuples (label, latex)
+#! Derived parameters should be followed by a *
+
+paramnames = [(f'p{i}', f'\\theta_{i}') for i in range(nDims)]
+paramnames += [('r*', 'r')]
+
+#| Run PolyChord
+
+output = pypolychord.run(
+    likelihood,
+    nDims,
+    nDerived=nDerived,
+    prior=prior,
+    dumper=dumper,
+    file_root='gaussian',
+    nlive=200,
+    do_clustering=True,
+    read_resume=False,
+    paramnames=paramnames,
+)
+
+#| Make an anesthetic plot 
+
+try:
+    import anesthetic as ac
+    fig, ax = ac.make_2d_axes(['p0', 'p1', 'p2', 'p3', 'r'])
```

Comment:
I'm not a fan of cluttering the namespace personally, and I thought that Lukas and you liked `ac` as a shorthand for anesthetic!

I'll settle for `import anesthetic` then `anesthetic.make_2d_axes`

Reply:
I've also imported numpy rather than clutter up the namespace with `log` and `pi`. Personally I've always found this annoying when I want to just substitute in a simple likelihood to check I haven't done anything daft, and copy-pasting the one from run_pypolychord.py is my go-to, but then I have to change `log`->`np.log` etc

Reply:
I’ll weigh in on the discussion. 

This is about consistency. If most of the project uses a “cluttered” namespace, it would be very distracting for users and other engineers. 

The “cluttering” of a namespace makes it easy to substitute different versions of similar API elements: constants, variables, functions etc.

Consider if you needed to replace `numpy.pi` with `scipy.pi`. We’re talking a single line change in the imports compared to every single instance having to be renamed. 

Another *huge* benefit IMO is readability. The namespace is not something the programmer sees. By contrast every place numpy functions or constants are used is by definition visible to the programmer. If there is no ambiguity, direct imports communicate more and make the program shorter. While `sys.os` functions may be harder to parse if imported directly, numpy and scipy functions are by definition known to the user.

Reply:
I can see the convenience of substituting similar api elements, however iirc with anesthetic we were once caught out by the differences between the numpy and scipy Cholesky decompositions when we substituted one for the other

I understand what you mean about readability, but then it also sometimes nice to be reminded where a function comes from

However I will choose my battles and for the sake of finally seeing the back of this PR, I've done as Will originally requested

Reply:
> However I will choose my battles and for the sake of finally seeing the back of this PR, I've done as Will originally suggested

Good choice. 

That's usually the way it goes when you're contributing. The maintainers set style and convention, and in the few situations where I've seen maintainers change their mind about either of those things, it was a separate discussion that came with its own PR. 

> I understand what you mean about readability, but then it's also sometimes nice to be reminded where a function came from without going back to the preamble

Very much up to the maintainer. It's a question of style, and each style has trade-offs. On that note, If I were curious about where a function came from, most all code editors (Emacs, Vim etc.) have a facility that will display the documentation for a symbol if you hover. There's also code lenses that will optionally qualify all functions regardless of the actual text of the program.

> I can see the convenience of substituting similar api elements, however iirc with anesthetic we were once caught out by the differences between the numpy and scipy Cholesky decompositions when we substituted one for the other

Sure. But that isn't exactly an argument for qualifying the paths.  

First, as a result of that debacle, you know that the two are not the same, and that you do need to disambiguate, this is valuable information if you plan to contribute/co-maintain a project. 

Secondly, there's more than one way to disambiguate; only one way to do it is to qualify the paths. Another way to do it, and I'll argue better, is to disambiguate by intent: `np.cholesky` → `upper_triangular`, and `scipy` → `lower_triangular`. Now  your code communicates information to the user, without them having to dig through documentation to find out that the difference is in the defaults.